include/simdjson/generic/ondemand/document_stream-inl.h

```diff
@@ -290,6 +293,8 @@ inline void document_stream::next_document() noexcept {
   if (error) { return; }
   // Always set depth=1 at the start of document
   doc.iter._depth = 1;
+  // consume comma if comma separated is allowed
+  if (allow_comma_separated) { doc.iter.consume_character(','); }
```

Comment:
I absolutely love that this is so confined to a single place! I'm a little surprised we couldn't use advance(), though, switching from thinking in terms of tokens (which `,` is one of) to thinking of characters does seem like a dangerous thing to me, even if it's confined to this one place where it works.

Reply:
@jkeiser I merged this because, as you have remarked, it is a very neat patch that is quite isolated.

We can change the design.

Are you saying that we should just skip over the token, no matter what it is? Ignoring its nature?

Can you elaborate on your concern?

(I am 100% open to changing this.)